doc/design/numeric_column_clause.md

```diff
@@ -0,0 +1,164 @@
+LABEL label
+INTO result_table;
+```
+
+In this document, we would discuss why and how we should redesign the numeric column clause in SQLFlow.
+
+
+## Background: Why Redesign the Numeric Column Clause
+
+Currently, SQLFlow uses the following methods to support loading data from a numeric column:
+
+- If each cell value is a single number, there are 3 equivalent ways to write the `TO TRAIN` SQL statement:
+    - Do not write the `COLUMN` clause.
+        ```sql
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        LABEL label
+        INTO result_table;
+        ```
+    - Write the column name directly in the `COLUMN` clause.
+        ```sql
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        COLUMN c1, c2
+        LABEL label
+        INTO result_table;
+        ```
+    - Write `NUMERIC()` or `NUMERIC(DENSE())` in the `COLUMN` clause.
+        ```sql
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        COLUMN NUMERIC(c1), NUMERIC(DENSE(c2))
+        LABEL label
+        INTO result_table;
+        ```
+        
+- If each cell value is a string that is encoded from a numeric vector, the decoded vector of the cell value may be dense or sparse. For example, the string `"1,3,5"` may represent (1) a dense vector `[1, 3, 5]`, or (2) a sparse vector `x` , in which `x[1] = x[3] = x[5] = 1` and the other values of `x` are all zeros. Users should specify whether the decoded vector is dense or sparse explicitly in the `COLUMN` clause.
+    - If the decoded vector is dense, there are 2 equivalent ways to write the `TO TRAIN` SQL statement:
+        - Use the `NUMERIC()` in the `COLUMN` clause and specify the length of the decoded vector in `NUMERIC()`.
+        ```sql
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        LABEL label
+        COLUMN NUMERIC(c1, 10), NUMERIC(c2, 10) -- 10 is the length of the decoded dense vector
+        INTO result_table;
+        ```
+        - Use the `NUMERIC(DENSE())` in the `COLUMN` clause and specify the length of the decoded vector in `DENSE()`.
+        ```sql
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        LABEL label
+        COLUMN NUMERIC(DENSE(c1, 10)), NUMERIC(DENSE(c2, 10)) -- 10 is the length of the decoded dense vector
+        INTO result_table;
+        ```
+    - If the decoded vector is sparse, there is only one way to write the `TO TRAIN` SQL statement.
+        ```sql
+        SELECT c1, c2, label FROM train_table
+        TO TRAIN DNNRegressor
+        LABEL label
+        COLUMN NUMERIC(SPARSE(c1, 10)), NUMERIC(SPARSE(c2, 10)) -- 10 is the dense length of the decoded sparse vector
+        INTO result_table;
+        ```
+
+There are some problems in the current design:
+
+- Both `NUMERIC` and `DENSE/SPARSE` have the `length/shape` parameter. 
```

Comment:
Do we support 2 or more dimension? If not, we may not need shape, length is enough

Reply:
Yes, we should support 2 or more dimensions.